pkg/reconciler/source/config_watcher_test.go

```diff
@@ -19,55 +19,98 @@ package source
 import (
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
 	"knative.dev/pkg/configmap"
-	. "knative.dev/pkg/reconciler/testing"
+	"knative.dev/pkg/logging"
+	loggingtesting "knative.dev/pkg/logging/testing"
+	"knative.dev/pkg/metrics"
+	tracingconfig "knative.dev/pkg/tracing/config"
 
 	_ "knative.dev/pkg/metrics/testing"
 )
 
-func TestConfigWatcher_defaults(t *testing.T) {
-	ctx, _ := SetupFakeContext(t)
-	cw := StartWatchingSourceConfigurations(ctx, "name", configmap.NewStaticWatcher(&corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "config-observability",
-			Namespace: "knative-eventing",
-		},
-		Data: map[string]string{
-			"_example": "test-config",
-		},
-	}, &corev1.ConfigMap{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "config-logging",
-			Namespace: "knative-eventing",
-		},
-		Data: map[string]string{
-			"_example": "test-config",
-		},
-	}, &corev1.ConfigMap{
+const testComponent = "test_component"
+
+func TestNewConfigWatcher_defaults(t *testing.T) {
```

Comment:
This test is good, but can you add again a test back with config maps only with example values? And add another one using options, always with config maps only with _example

Reply:
Will do.

Reply:
Done in e640e85